TrafficTracker.py

```diff
@@ -41,38 +49,46 @@ def run(self, video_path, start_time=time.time()):
         _, out = self.initialize_video_writer(
             opt.output_path, self.vid_fps, self.vid_width, self.vid_height)
         _, og_frame = self.vid.read()  # BGR
-
+        
         metrics = []
+
+        pbar = tqdm(total=self.length+1)
+
+                
         while og_frame is not None:
-            new_frame = self.preprocess_image(og_frame, image_size)
+            pbar.update()
+            new_frame = self.preprocess_image(og_frame, opt.image_size)
 
             boxes, class_inds, scores = yolo_predict(
-                self.yolo, new_frame, og_frame)
-            boxes = np.array([list(box) for box in boxes])
-            names = [self.class_names[name] for name in class_inds]
-            outputs = self.deepsort.update(boxes, names, scores, og_frame)
-
-            # Generate Metrics
-            frame_time = self.get_next_time(frame_time, self.vid_fps)
-
-            for i, _ in enumerate(outputs):
-                outputs[i][0:4] = self.xywh_to_tlwh(outputs[i][0:4])
-                metrics.append({
-                    "created_time": frame_time,
-                    "Pos_x": outputs[i][0],
-                    "Pos_y": outputs[i][1],
-                    "width": outputs[i][2],
-                    "height": outputs[i][3],
-                    "Class": outputs[i][5],
-                    "Object_id": outputs[i][4],
-                    "location_id": "McMaster University"
-                })
-
-            if len(outputs) > 0:
-                bbox_tlwh = outputs[:, :4]
-                identities = outputs[:, 4]
-                og_frame = self.draw_boxes(og_frame, bbox_tlwh, identities)
-                out.write(og_frame)
+                self.yolo, new_frame, og_frame,self.log)    
+            # checks if there is an output, otherwise just keeps the original frame
+            if boxes and class_inds and scores:
```

Comment:
`if boxes and class_inds and scores`

this checks if it received non-empty response for that frame, if there is nothing, don't process frame